lib/index.js

```diff
@@ -1,3 +1,232 @@
 'use strict';
 
-module.exports = () => true;
+const utils = require('./utils');
+
+/**
+ * TODO: describe the backwards compatibility
+ * @type {string}
+ */
+const BACKWARD = 'BACKWARD';
+
+/**
+ * TODO: describe the forward compatibility
+ * @type {string}
+ */
+const FORWARD = 'FORWARD';
+
+/**
+ * TODO: describe full compatibility
+ * @type {string}
+ */
+const FULL = 'FULL';
+
+/**
+ * The list of schema property names to ignore
+ * @type {string[]}
+ */
+const IGNORED_PROPERTIES = [ 'description', '$comment' ];
+
+/**
+ * A collection of checkers for each of the types supported by JSON schema.
+ * @type {{string: typeCheckers.string, object: typeCheckers.object}}
+ */
+
+const typeCheckers = {
+    array: (oldSubSchema, newSubSchema, context) => {
+        utils.verifySchemasType('array', oldSubSchema, newSubSchema);
+        return [];
```

Comment:
I have a hard time understanding why all of the methods here either generate an error or return an empty array. Shouldn't it return the diffs so that we can accumulate them and present the errors to the user?